### ISSUE TITLE
Output buffer is full

### ISSUE THREAD

**Issue by psy-q** [joshkunz/ashuffle#9]
When using ashuffle on a collection of about 80,000 songs on a Raspi with 512 MB RAM, I got the following error in mpd.log:

```
May 27 17:29 : buffered_socket: error on client 2: Output buffer is full
```

And ashuffle says:

```
MPD error: Connection closed by the server
```

Is the collection too big? The Raspi's memory too small? Both? :)

**Comment by joshkunz** [comment]
tl;dr just set `max_output_buffer_size 30720` in your `~/.mpd.conf`.

ashuffle basically needs enough memory to store a list of the paths (relative to your music directory) to every file in your library. The longest such path in my library is about 210 characters, so even if every path in your library was as long as that path, it would still only take about 16MB to store it all.

Given this error:
> `May 27 17:29 : buffered_socket: error on client 2: Output buffer is full`

It looks like you're probably running into MPDs response size limit. ashuffle needs a complete list of your songs to work properly, and MPD by default limits the size of a response to 8MB. If your library is bigger than that, it will trigger that `buffer is full` error and fail.

To increase the buffer size, set the `max_output_buffer_size KBYTES` [option](https://www.musicpd.org/doc/user/config_other.html) in your `~/.mpd.conf` file. Given those rough calculations, 30MB should be plenty, so just add

    max_output_buffer_size 30720

Unless your song file names are *really* long, this should work fine for you.

**Comment by psy-q** [comment]
Thanks a lot for the detailed explanation! I tried this and first ashuffle reported a timeout from MPD after a minute or so, but the second time it seems to have worked. Maybe a Raspi isn't the best thing to use for 80k songs, either. I'll be thinking about an upgrade as well if I run into issues.

Cheers and thanks again!
